src/ssl.c

```diff
@@ -42,11 +43,18 @@
 #ifdef HAVE_SSL
 
 #include <openssl/err.h>
-#include <openssl/x509.h>
+#include <openssl/x509v3.h>
 
-#ifndef X509_CHECK_FLAG_NO_PARTIAL_WILDCARDS
-#define X509_CHECK_FLAG_NO_PARTIAL_WILDCARDS 4
-#endif
+typedef enum {
+	MatchFound,
+	MatchNotFound,
+	NoSANPresent,
+	MalformedCertificate,
+	Error
+} validate_result;
```

Comment:
Is this necessary?

Reply:
It's helpful to be explicit within the private verify implementation. If you have an alternate suggestion I am happy to refactor.